superset/security/manager.py

```diff
@@ -877,3 +878,39 @@ def assert_viz_permission(self, viz: "BaseViz") -> None:
         """
 
         self.assert_datasource_permission(viz.datasource)
+
+    def get_rls_filters(self, table):
+        """
+        Retrieves the appropriate row level security filters for the current user and the passed table.
+
+        :param table: The table to check against
+        :returns: A list of clause strings.
+        """
+        try:
+            roles = [role.id for role in g.user.roles]
+            return [
+                f.clause
+                for f in table.row_level_security_filters
+                if any(r.id in roles for r in f.roles)
+            ]
+        except AttributeError:
+            return []
+
+    def get_rls_ids(self, table) -> List[int]:
+        """
+        Retrieves the appropriate row level security filters IDs for the current user and the passed table.
+
+        :param table: The table to check against
+        :returns: A list of IDs.
+        """
+        try:
+            roles = [role.id for role in g.user.roles]
+            ids = [
+                f.id
+                for f in table.row_level_security_filters
+                if any(r.id in roles for r in f.roles)
+            ]
+            ids.sort()  # Combinations rather than permutations
+            return ids
+        except AttributeError:
```

Comment:
Hmm, I know it's a bit more verbose, but I prefer checking for the existence of the attribute using `hasattr` rather than catching the `AttributeError`.

Reply:
Isn't `try/except` more "pythonic"?  I have no problem changing this, but worry about someone else then requesting it be changed back.  Can we get a second opinion on this one?

Reply:
I agree with @serenajiang here; best to be explicit about the attribute we`re after.

Reply:
done